http/src/request/application/interaction/create_followup_message.rs

```diff
@@ -293,37 +243,53 @@ impl TryIntoRequest for CreateFollowupMessage<'_> {
             webhook_id: self.application_id.get(),
         });
 
-        if !self.attachments.is_empty() || self.fields.payload_json.is_some() {
-            let mut form = Form::new();
-
-            if !self.attachments.is_empty() {
-                for (index, attachment) in self.attachments.iter().enumerate() {
-                    form.attach(
-                        index as u64,
-                        attachment.filename.as_bytes(),
-                        attachment.file,
-                    );
-                    self.fields.attachments.push(PartialAttachment {
-                        id: index as u64,
-                        filename: attachment.filename,
-                        description: attachment.description,
-                    })
-                }
+        // Interaction executions don't need the authorization token, only the
+        // interaction token.
+        request = request.use_authorization_token(false);
+
+        // Set the default allowed mentions if required.
+        if self.fields.allowed_mentions.is_none() {
+            if let Some(allowed_mentions) = self.http.default_allowed_mentions() {
+                self.fields.allowed_mentions = Some(NullableField(Some(allowed_mentions)));
+            }
+        }
+
+        // Determine whether we need to use a multipart/form-data body or a JSON
+        // body.
+        if self.attachment_files.is_some() || self.fields.payload_json.is_some() {
+            if let Some(attachment_files) = self.attachment_files {
+                self.fields.attachments = Some(
+                    attachment_files
+                        .iter()
+                        .enumerate()
+                        .map(|(index, attachment)| PartialAttachment {
+                            description: attachment.description,
+                            filename: Some(attachment.filename),
+                            id: index as u64,
+                        })
+                        .collect(),
+                );
             }
 
-            if let Some(payload_json) = &self.fields.payload_json {
-                form.payload_json(payload_json);
+            let mut form_builder = if let Some(payload_json) = self.fields.payload_json {
+                FormBuilder::new(Cow::Borrowed(payload_json))
             } else {
-                let body = crate::json::to_vec(&self.fields).map_err(HttpError::json)?;
-                form.payload_json(&body);
+                crate::json::to_vec(&self.fields)
+                    .map(Cow::Owned)
+                    .map(FormBuilder::new)
+                    .map_err(HttpError::json)?
+            };
+
+            if let Some(attachment_files) = self.attachment_files {
+                form_builder = form_builder.attachments(attachment_files);
             }
 
-            request = request.form(form);
+            request = request.form(form_builder.build());
```

Comment:
i have extracted it some, but i don't know how far i can go without making it too complex. imo it would be better to show the logic in try_into_request than hide it